### PR TITLE
Add documentations for Asgardeo Trigger.

### PR DESCRIPTION
**Pull request by UdeshAthukorala**

### CODE REVIEW

File: asyncapi/asgardeo/Module.md [hunk 1]
```diff
@@ -1,6 +1,66 @@
 ## Overview
 
-The `ballerinax/trigger.asgardeo` module provides a Listener to grasp events triggered from your Asgardeo App. This functionality is provided by Asgardeo Events API.
+The `ballerinax/trigger.asgardeo` module provides a Listener to grasp events triggered from your [Asgardeo](https://wso2.com/asgardeo/) Organization. This functionality is provided by Asgardeo Events API.
```
Comment (nilminiwso2) [inline asyncapi/asgardeo/Module.md:3]:
```suggestion
The `ballerinax/trigger.asgardeo` module provides a Listener to grasp events triggered from your [Asgardeo](https://wso2.com/asgardeo/) organization. This functionality is provided by Asgardeo Events API.
```

File: asyncapi/asgardeo/Module.md [hunk 2]
```diff
@@ -1,6 +1,66 @@
...
+
+The Asgardeo Trigger module allows you to listen to following events occur in your Asgardeo organization.
```
Comment (nilminiwso2) [inline asyncapi/asgardeo/Module.md:5]:
```suggestion
The Asgardeo Trigger module allows you to listen to the following events when they occur in your Asgardeo organization:
```

File: asyncapi/asgardeo/Module.md [hunk 3]
```diff
@@ -1,6 +1,66 @@
...
+- An organization in Choreo created with the same email address and under the same name that was used to create the organization in Asgardeo.
+- Enable event publishing in your Asgardeo Organization.
 
 ## Quickstart
-To use the Asgardeo listener, please go the [Choreo Console](https://console.choreo.dev) and create a webhook
\ No newline at end of file
+
+### Step 1: Create a webhook in Choreo to receive Asgardeo events
+
+- To use the Asgardeo listener, please go the [Choreo Console](https://console.choreo.dev) and create a webhook.
```
Comment (nilminiwso2) [inline asyncapi/asgardeo/Module.md:29]:
```suggestion
To use the Asgardeo listener, go to the [Choreo Console](https://console.choreo.dev) and create a webhook.
```

File: asyncapi/asgardeo/Module.md (hunk 1 above)
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Module.md]: Addressed via https://github.com/ballerina-platform/asyncapi-triggers/pull/113/commits/2ad7204cdcd395c42e6c4985b8b686f2296ed0e2

File: asyncapi/asgardeo/Module.md (hunk 2 above)
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Module.md]: Addressed via https://github.com/ballerina-platform/asyncapi-triggers/commit/2ad7204cdcd395c42e6c4985b8b686f2296ed0e2

**Review by HasithaAthukorala — DISMISSED** [review]

File: asyncapi/asgardeo/Module.md [hunk 4]
```diff
@@ -1,6 +1,70 @@
...
\ No newline at end of file
+
+### Step 1: Create a webhook in Choreo to receive Asgardeo events
+
+To use the Asgardeo listener, go to the [Choreo Console](https://console.choreo.dev) and create a webhook  with **Asgardeo** as the trigger type.
+Learn more about how to develop a webhook from [here.](https://wso2.com/choreo/docs/webhook/#develop-a-webhook)
+
+### Step 2: Program a logic to trigger for Asgardeo events
+
+- Now you can implement listener remote functions supported by this connector.
```
Comment (nilminiwso2) [inline asyncapi/asgardeo/Module.md:37]:
```suggestion
Now you can implement listener remote functions supported by this connector.
```

File: asyncapi/asgardeo/Module.md [hunk 5]
```diff
@@ -1,6 +1,70 @@
...
+- Write a remote function to receive a particular event type.
```
Comment (nilminiwso2) [inline asyncapi/asgardeo/Module.md:38]:
```suggestion
1. Write a remote function to receive a particular event type. 
```

File: asyncapi/asgardeo/Module.md [hunk 6]
```diff
@@ -1,6 +1,70 @@
...
+### Step 1: Create a webhook in Choreo to receive Asgardeo events
+
+To use the Asgardeo listener, go to the [Choreo Console](https://console.choreo.dev) and create a webhook  with **Asgardeo** as the trigger type.
+Learn more about how to develop a webhook from [here.](https://wso2.com/choreo/docs/webhook/#develop-a-webhook)
+
+### Step 2: Program a logic to trigger for Asgardeo events
+
+- Now you can implement listener remote functions supported by this connector.
+- Write a remote function to receive a particular event type. 
+- Implement your logic within that function as shown in the below sample. This is an example of logging Asgardeo events in the Choreo console.
```
Comment (nilminiwso2) [inline asyncapi/asgardeo/Module.md:39]:
```suggestion
2. Implement your logic within that function as shown in the below sample. This is an example of logging Asgardeo events in the Choreo console.
```

File: asyncapi/asgardeo/Module.md [hunk 7]
```diff
@@ -1,6 +1,70 @@
...
+        remote function onAcceptUserInvite(asgardeo:GenericEvent event ) returns error? {
+            log:printInfo(event.toJsonString());
+        }
+    }
+    
+    service /ignore on httpListener {}
+    ```
+
+### Step 3: Deploy the Webhook in Choreo
+- Then navigate to the deploy tab and click on the Config & Deploy button to start the deploy process. Learn more about how to deploy a webhook from [here.](https://wso2.com/choreo/docs/webhook/#deploy-a-webhook)
```
Comment (nilminiwso2) [inline asyncapi/asgardeo/Module.md:70]:
```suggestion
Once you have created the webhook, go to the **Deploy** tab on the Choreo Console and click on **Config & Deploy** to start the deployment process. Learn more about how to deploy a webhook from [here](https://wso2.com/choreo/docs/webhook/#deploy-a-webhook).
```

File: asyncapi/asgardeo/Module.md [hunk 8]
```diff
@@ -1,6 +1,70 @@
...
+- Enable event publishing in your Asgardeo Organization. Learn how to enable from [here.](https://wso2.com/asgardeo/docs/references/asgardeo-events/#configure-asgardeo-to-publish-events)
 
 ## Quickstart
-To use the Asgardeo listener, please go the [Choreo Console](https://console.choreo.dev) and create a webhook
\ No newline at end of file
+
+### Step 1: Create a webhook in Choreo to receive Asgardeo events
+
+To use the Asgardeo listener, go to the [Choreo Console](https://console.choreo.dev) and create a webhook  with **Asgardeo** as the trigger type.
+Learn more about how to develop a webhook from [here.](https://wso2.com/choreo/docs/webhook/#develop-a-webhook)
```
Comment (nilminiwso2) [inline asyncapi/asgardeo/Module.md:33]:
```suggestion
Learn more about [how to develop a webhook](https://wso2.com/choreo/docs/webhook/#develop-a-webhook).
```

File: asyncapi/asgardeo/Module.md [hunk 9]
```diff
@@ -1,6 +1,70 @@
...
+
+## Prerequisites
+- An organization in Asgardeo. Learn how to sign up and create an organization from [here](https://wso2.com/asgardeo/docs/get-started/create-asgardeo-account/#sign-up).
+- An organization in [Choreo](https://wso2.com/choreo/) created with the same email address and under the same name that was used to create the organization in Asgardeo.
+- Enable event publishing in your Asgardeo Organization. Learn how to enable from [here.](https://wso2.com/asgardeo/docs/references/asgardeo-events/#configure-asgardeo-to-publish-events)
 
 ## Quickstart
-To use the Asgardeo listener, please go the [Choreo Console](https://console.choreo.dev) and create a webhook
\ No newline at end of file
+
```
Comment (nilminiwso2) [inline asyncapi/asgardeo/Module.md:29]:
```suggestion
Follow the steps given below to try out the Asgardeo listener.
```

File: asyncapi/asgardeo/Module.md [hunk 10]
```diff
@@ -1,6 +1,70 @@
 ## Overview
 
-The `ballerinax/trigger.asgardeo` module provides a Listener to grasp events triggered from your Asgardeo App. This functionality is provided by Asgardeo Events API.
+The `ballerinax/trigger.asgardeo` module provides a Listener to grasp events triggered from your [Asgardeo](https://wso2.com/asgardeo/) organization. This functionality is provided by the [Asgardeo Events API.](https://wso2.com/asgardeo/docs/references/asgardeo-events/)
```
Comment (nilminiwso2) [inline asyncapi/asgardeo/Module.md:3]:
```suggestion
The `ballerinax/trigger.asgardeo` module provides a Listener to grasp events triggered from your [Asgardeo](https://wso2.com/asgardeo/) organization. This functionality is provided by the [Asgardeo Events API](https://wso2.com/asgardeo/docs/references/asgardeo-events/).
```

File: asyncapi/asgardeo/Module.md [hunk 11]
```diff
@@ -1,6 +1,70 @@
...
+
+The Asgardeo Trigger module allows you to listen to the following events when they occur in your Asgardeo organization:
+
+- Registration Events
```
Comment (nilminiwso2) [inline asyncapi/asgardeo/Module.md:7]:
```suggestion
- **Registration Events**
```

Comment (nilminiwso2) [inline asyncapi/asgardeo/Module.md:7]:
```suggestion
- **Registration Events**
```

File: asyncapi/asgardeo/Module.md [hunk 12]
```diff
@@ -1,6 +1,70 @@
...
+   - Add User Event
+   - Accept User Invite Event
+   - Confirm Self Sign Up Event
+- User Operation Events
```
Comment (nilminiwso2) [inline asyncapi/asgardeo/Module.md:11]:
```suggestion
- **User Operation Events**
```

File: asyncapi/asgardeo/Module.md [hunk 13]
```diff
@@ -1,6 +1,70 @@
...
+   - User Account Lock Event
+   - User Account Unlock Event
+   - User Credential Update Event
+   - User Delete Event
+   - User Group Update Event
+- Login Events
```
Comment (nilminiwso2) [inline asyncapi/asgardeo/Module.md:17]:
```suggestion
- **Login Events**
```

File: asyncapi/asgardeo/Module.md [hunk 14]
```diff
@@ -1,6 +1,70 @@
...
+   - Login Success Event
+
+Learn more about Asgardeo events from [here.](https://wso2.com/asgardeo/docs/references/asgardeo-events/#configure-asgardeo-to-publish-events)
```
Comment (nilminiwso2) [inline asyncapi/asgardeo/Module.md:20]:
```suggestion
Learn more about [Asgardeo events](https://wso2.com/asgardeo/docs/references/asgardeo-events/#configure-asgardeo-to-publish-events).
```

File: asyncapi/asgardeo/Package.md [hunk 15]
```diff
@@ -1,7 +1,7 @@
-Listen to Asgardeo Events API from Ballerina
+Listen to [Asgardeo Events API](https://wso2.com/asgardeo/docs/references/asgardeo-events/) from Ballerina
 
 ## Package overview
-The `ballerinax/trigger.asgardeo` is a [Ballerina](https://ballerina.io/) trigger for Asgardeo.
+The `ballerinax/trigger.asgardeo` is a [Ballerina](https://ballerina.io/) trigger for [Asgardeo](https://wso2.com/asgardeo/).
```
Comment (nilminiwso2) [inline asyncapi/asgardeo/Package.md:4]: Under Compatibility, the Slack link should be changed to Discord.

File: asyncapi/asgardeo/Module.md (hunk 4 above)
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Module.md]: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/pull/113/commits/fcd92a802f674aec9896fac8bc36af872d2f40c5

File: asyncapi/asgardeo/Module.md (hunk 6 above)
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Module.md]: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

File: asyncapi/asgardeo/Module.md (hunk 8 above)
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Module.md]: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

File: asyncapi/asgardeo/Module.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Module.md:28]: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

File: asyncapi/asgardeo/Module.md (hunk 10 above)
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Module.md]: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

File: asyncapi/asgardeo/Module.md (hunk 11 above)
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Module.md]: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

File: asyncapi/asgardeo/Module.md (hunk 12 above)
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Module.md]: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

File: asyncapi/asgardeo/Module.md (hunk 14 above)
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Module.md]: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

File: asyncapi/asgardeo/Package.md (hunk 15 above)
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Package.md:4]: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

File: asyncapi/asgardeo/Module.md (hunk 13 above)
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Module.md]: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

File: asyncapi/asgardeo/Module.md (hunk 11 above)
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Module.md]: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

File: asyncapi/asgardeo/Module.md (hunk 9 above)
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Module.md:28]: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

File: asyncapi/asgardeo/Module.md (hunk 7 above)
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Module.md]: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

File: asyncapi/asgardeo/Module.md (hunk 5 above)
Comment (UdeshAthukorala) [reply asyncapi/asgardeo/Module.md]: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

**Review by HasithaAthukorala — APPROVED** [review]

**Pull request merged**